salt/pillar/vmware_pillar.py

```diff
@@ -0,0 +1,524 @@
+
+The pillar will return an empty dict if the 'os' or 'virtual' grain are not 'VMWare', 'ESXi', or 'VMWare ESXi'.
+
+Defaults
+========
+    The external pillar will search for Virtual Machines with the VM name matching the minion id.
+    Data will be returned into the 'vmware' pillar key.
+    The external pillar has a default set of properties to return for both VirtualMachine and HostSystem types.
+
+
+Configuring the VMWare pillar
+============================
+
+The required minimal configuration in the salt master ext_pillar setup:
+
+.. code-block:: yaml
+
+    ext_pillar:
+        - vmware:
+            host: <vcenter/esx host>
+            username: <user to connect with>
+            password: <password>
+
+Optionally, the following keyword arguments can be passed to the ext_pillar for customized configuration:
+
+    pillar_key
+        Optionally set the pillar key to return the data into.  Default is ``vmware``.
+
+    protocol
+        Optionally set to alternate protocol if the vCenter server or ESX/ESXi host is not
+        using the default protocol. Default protocol is ``https``.
+
+    port
+        Optionally set to alternate port if the vCenter server or ESX/ESXi host is not
+        using the default port. Default port is ``443``.
+
+    property_name
+        Property name to match the minion id against.  Defaults to ``name``.
+
+    property_types
+        Optionally specify a list of pyVmomi vim types to search for the minion id in 'property_name'.
+        Default is ``['VirtualMachine']``.
+
+        For example, to search both vim.VirtualMachine and vim.HostSystem object types:
+
+        .. code-block:: yaml
+
+           ext_pillar:
+               - vmware:
+                   host: myesx
+                   username: root
+                   password: complex_password
+                   property_types:
+                     - VirtualMachine
+                     - HostSystem
+
+        Additionally, the list of property types can be dicts, the item of the dict being a list specifying
+        the attribute to return for that vim object type.
+
+        The pillar will attempt to recurse the attribute and return all child attributes.
+
+        To explicitly specify deeper attributes without attempting to recurse an attribute, convert the list
+        item to a dict with the item of the dict being the child attributes to return.  Follow this pattern
+        to return attributes as deep within the object as necessasry.
+
+        .. note::
+            Be careful when specifying custom attributes!  Many attributes have objects as attributes which
+            have the parent object as an attribute and which will cause the pillar to fail due to the attempt
+            to convert all sub-objects recursively (i.e. infinite attribute loops).  Specifying only the
+            sub-attributes you would like returned will keep the infinite recursion from occurring.
```

Comment:
A maximum recursion depth exceeded exception occurs and the pillar does not return.  I added a blurb in the doc string and corrected the spelling mistake.